IABTechLab/uid2-self-serve-portal
# Upgrade Keycloak to 23.0.0

- Upgrade Keycloak to 23.0.0
- Update dependencies and docker yaml so that it actually works
- Style the show/hide password toggle that comes for free with version 23.
- Refactor `Keycloak.ts` to ensure tests in `PortalErrorBounday.test.tsx` do not fail with `Error: A 'Keycloak' instance can only be initialized once.`

## Bug fix
**Before**
If you login with user1, then successfully login with user2 while user1's session is still active, the login page will tell you you're already logged in and link you to the portal. But even though you just successfully authenticated with user2, your session will be for user1.

**After**

https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/cd52d17c-64ae-4c32-b2cc-b06c47cb643e

## Show/hide password toggle

https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/f22fbdea-36c3-431c-b607-5654b09e6e50

**Updated hover styling**


https://github.com/IABTechLab/uid2-self-serve-portal/assets/137864392/50144c53-7956-4fea-925f-4bf181db5a29

## Manual testing
1. Check out this branch
2. Run `npm i`
3. Re-run docker command to run Keycloak on the updated version
4. Perform manual tests such as logging in, trying to login as user2 while user1's session is still active, logging out ,etc.
